src/lib/accessibility/Behaviors/List/SelectableListItemBehavior.ts

```diff
@@ -11,6 +12,23 @@ const SelectableListItemBehavior: (props: any) => IAccessibilityDefinition = (pr
     root: {
       role: 'option',
       'aria-selected': !!props['active'],
+      tabIndex: props.atomicItemProps.isFocused ? '0' : '-1',
+    },
+  },
+  keyActions: {
```

Comment:
did you try moving the keyActions to the container? it looks like that might simplify the logic and reduce the number of callbacks sent to the item as this is updating the container state anyway